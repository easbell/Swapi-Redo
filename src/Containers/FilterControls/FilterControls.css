.filter {
  margin: 20px;
}

.filter button {
  border-radius: 5px;
  font-size: 20px;
  padding: 14px;
  margin: 5px 33px;
  background-color: #545558;
  color: white;
  border: 0;
}

.filter button:hover {
  background-color: #fff;
  color: #545558;
  cursor: pointer;
}