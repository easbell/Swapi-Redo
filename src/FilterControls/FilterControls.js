import React from 'react';
import { connect } from 'react-redux';
import { selectCategory } from '../actions';
import './FilterControls.css'

const FilterControls = (props) => {
  const handleSelection = (e) => {
    props.selectCategory(e.target.className);
  }

  return (
    <div className='filter'>
      <button onClick={handleSelection} className='people'>People</button>
      <button onClick={handleSelection} className='planets'>Planets</button>
      <button onClick={handleSelection} className='vehicles'>Vehicles</button>
    </div>
  )

  
}

export const mapDispatchToProps = (dispatch) => ({
  selectCategory: (category) => dispatch(selectCategory(category))
})

export default connect(null, mapDispatchToProps)(FilterControls);