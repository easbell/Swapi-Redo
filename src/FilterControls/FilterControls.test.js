import React from 'react';
import { FilterControls, mapDispatchToProps} from './FilterControls';
import { mount, shallow } from 'enzyme';
import { connect } from 'react-redux';

describe('FilterControls', () => {
  let wrapper;
  let mockSelectCategory;
  let mockHandleSelection;

  beforeEach(() => {
    mockSelectCategory = jest.fn();
    mockHandleSelection = jest.fn();
    wrapper = mount(<FilterControls selectCategory={mockSelectCategory}/>)
  });

  it('should match snapshot with all data passed in', () => {
    wrapper = shallow(<FilterControls selectCategory={mockSelectCategory}/>)
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should invoke handleSelection with proper argument when button is clicked', () => {
    const mockedEvent = { target: {} }
    wrapper.find('.people').simulate('click', mockedEvent)
    expect(mockHandleSelection).toHaveBeenCalled()
  });


})
