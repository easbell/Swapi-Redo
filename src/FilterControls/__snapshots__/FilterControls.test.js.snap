// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterControls should match snapshot with all data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterControls
    selectCategory={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="people"
          onClick={[Function]}
        >
          People
        </button>,
        <button
          className="planets"
          onClick={[Function]}
        >
          Planets
        </button>,
        <button
          className="vehicles"
          onClick={[Function]}
        >
          Vehicles
        </button>,
      ],
      "className": "filter",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "className": "people",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "className": "planets",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "className": "vehicles",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="people"
            onClick={[Function]}
          >
            People
          </button>,
          <button
            className="planets"
            onClick={[Function]}
          >
            Planets
          </button>,
          <button
            className="vehicles"
            onClick={[Function]}
          >
            Vehicles
          </button>,
        ],
        "className": "filter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": "people",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": "planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "className": "vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
